server-core/service/src/main/java/io/infectnet/server/service/impl/UserServiceImpl.java

```diff
@@ -26,80 +33,119 @@ public UserServiceImpl(){
     }
 
     public UserServiceImpl(UserStorage userStorage, TokenStorage tokenStorage, ConverterServiceImpl converterService) {
+
         this.userStorage = userStorage;
         this.tokenStorage = tokenStorage;
         this.converterService = converterService;
     }
 
-    /**
-     * Registers the new user in. Checks if the token has not expired yet and the username is unique,
-     * and whether or not the password is valid according to the specified rules:
-     *      minimum length of 8 characters
-     *      must contain at least one number
-     *      must contain at least one uppercase letter.
-     * If authentication fails no new user will be stored, otherwise a new User is stored
-     * with the username, the email, the password and the date of the registration now produced.
-     *
-     * @param token the token from the current registration
-     * @param email the email address from the current registration
-     * @param username the username from the current registration
-     * @param password the password from the current registration
-     * @return the Optional containing the UserDTO made from the created user
-     */
     @Override
     public Optional<UserDTO> register(String token, String email, String username, String password) {
-        if(tokenIsValid(token) && emailIsUnique(email) && usernameIsUnique(username) && passwordIsValid(password)){
-            UserDTO newUser = new UserDTO(username,email,password, LocalDateTime.now());
+
+        if(isRegisterValid( Objects.requireNonNull(token),
+                            Objects.requireNonNull(email),
+                            Objects.requireNonNull(username),
+                            Objects.requireNonNull(password))
+                ){
+
+            UserDTO newUser =
+                    new UserDTO(Objects.requireNonNull(username),
```

Comment:
`requireNonNull` calls are not necessary anymore here. You can safely remove them.